### PR TITLE
Jetpack Manage: new navigation UI enhancements

### PR DESCRIPTION
**Pull request by yashwin**
Resolves https://github.com/Automattic/jetpack-genesis/issues/72

## Proposed Changes

This PR does the below UI enhancements:

- Remove border-bottom for Sites Management if new navigation is enabled.
- Remove border-bottom for Plugin Management if new navigation is enabled.
- Remove border-bottom for Licenses if new navigation is enabled.
- Update the `Licenses` icon to be flat instead of filled.
- Update icon colors for Jetpack Cloud
- Update menu item font size to 13px

### Testing Instructions

**Prerequisites**

Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

**Instructions**

1. Open the Jetpack Cloud live link
2. Visit the /dashboard & append the URL with `?flags=jetpack/new-navigation`
3. Verify that you don't see the border-bottom for the page header. Verify that the `Licenses` icon looks flat instead of filled. Make sure to test the changes without the flags and see there are not issues.

Screenshot 2023-10-24 at 3 12 24 PM

4. Click Plugin Management > Verify that you don't see the border-bottom for the page header.

Screenshot 2023-10-24 at 3 15 42 PM

5. Click Licenses > Verify that you don't see the border-bottom for the page header.

Screenshot 2023-10-24 at 3 15 55 PM

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by ilonagl — APPROVED** [review]
Looks great!

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/style.scss [hunk 1]
```diff
@@ -41,6 +41,10 @@
 	@include breakpoint-deprecated( ">660px" ) {
 		border-bottom: 1px solid var(--color-primary-5);
 	}
+
+	&.is-new-navigation {
+		border-bottom: none;
```
Comment (jkguidaven) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/style.scss]: We can use RTL-friendly styling here.

```
border-block-end: none;
```

File: client/my-sites/plugins/style.scss [hunk 2]
```diff
@@ -454,6 +454,10 @@ body.is-section-plugins #primary {
 	@include break-large() {
 		padding: 6px 48px 0;
 	}
+
+	&.is-new-navigation {
+		border-bottom: none;
```
Comment (jkguidaven) [inline client/my-sites/plugins/style.scss]: Same here. we can use RTL-friendly css.

**Review by jkguidaven — COMMENTED** [review]
It seems that the top margin for the Licenses table is gone. 

**Now**
Screen Shot 2023-10-24 at 6 21 43 PM

**Before**
Screen Shot 2023-10-24 at 6 21 33 PM

I also noticed another issue we need to fix. The lower part of the content has different colors. I assume it was because the height of our content was less than 100% to account for the header. But now it was gone, this gap appeared.

277619646-0618a8ae-34a7-4aa9-b7cf-df08a74bd12e

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/style.scss (hunk 1 above)
Comment (yashwin) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/style.scss]: Done ✅

File: client/my-sites/plugins/style.scss (hunk 2 above)
Comment (yashwin) [reply client/my-sites/plugins/style.scss]: Done ✅

**Comment by yashwin** [comment]
Thanks for the thorough review, @jkguidaven! 

I have fixed these issues: https://github.com/Automattic/wp-calypso/pull/83409/commits/28a51aea367ca0edffbc47822c47f082f0b1dba7

Could you please take another look? TIA 🙂

**Review by jkguidaven — APPROVED** [review]
Thanks for addressing my comments. LGTM 👍

**Pull request merged**